### PR TITLE
Add documentation `Using Nx`

### PR DESCRIPTION
**Pull request by tschaffter**
Provides information one how to use Nx. This includes the definition of naming conventions and how to structure the workspace.

### CODE REVIEW

File: docs/using-nx.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+# Using Nx
+
+## Developer Workflow
```
[the remark this answers is not in the document]
Comment (tschaffter) [reply docs/using-nx.md:3]: Done

**Review by vpchung — APPROVED** [review]
Looks good for the most part.  Just a couple suggestions/clarifications.

File: docs/using-nx.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+mid-size applications.
+
+For example:
+
+```
+apps/web-app      <---- app named `web-app`
+libs/web          <---- app-specific libraries
+```
+
+Note: We would probably rename the above app `challenge-registry` the day this
```
[the remark this answers is not in the document]
Comment (tschaffter) [reply docs/using-nx.md:73]: Done

File: docs/using-nx.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+Since Nx allows us to group apps and libs in directories, those directories can
+become code-ownership boundaries. That's why the structure of an Nx workspace
+often reflects the structure of an organization. GitHub users can use the
+CODEOWNERS file for that.
+
+```
+/libs/happynrwlapp          julie-happynrwlapp-lead
+/apps/happynrwlapp          julie-happynrwlapp-lead
+/libs/shared/ui             hank-the-ui-guy
+/libs/shared/utils-testing  julie,hank
```
[the remark this answers is not in the document]
Comment (tschaffter) [reply docs/using-nx.md:125]: Now using only `julie` and `hank`.

**Pull request merged**